src/queue/extras.rs

```diff
@@ -1,12 +1,58 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
 
+use super::multilevel::ElapsedTime;
+
+use rand::prelude::*;
+use std::sync::Arc;
 use std::time::Instant;
 
-/// The extras for the task cells pushed into a simple task queue.
-///
-/// [`Default::default`] can be used to create a [`Extras`] for a task cell.
-#[derive(Debug, Default, Clone, Copy)]
+/// The extras for the task cells pushed into a task queue.
+#[derive(Debug, Clone)]
 pub struct Extras {
     /// The instant when the task cell is pushed to the queue.
-    pub schedule_time: Option<Instant>,
+    pub(crate) schedule_time: Option<Instant>,
+    /// The identifier of the task. Only used in the multilevel task queue.
+    pub(crate) task_id: u64,
+    /// The time spent on handling this task. Only used in the multilevel task
+    /// queue.
+    pub(crate) running_time: Option<Arc<ElapsedTime>>,
+    /// The level of queue which this task comes from. Only used in the
+    /// multilevel task queue.
+    pub(crate) current_level: u8,
+    /// If `fixed_level` is `Some`, this task is always pushed to the given
+    /// level. Only used in the multilevel task queue.
+    pub(crate) fixed_level: Option<u8>,
+}
+
+impl Extras {
+    /// Creates a default `Extras` for task cells pushed into a simple task
+    /// queue.
+    pub fn simple_default() -> Extras {
+        Extras {
+            schedule_time: None,
+            task_id: 0,
+            running_time: None,
+            current_level: 0,
+            fixed_level: None,
+        }
+    }
+
+    /// Creates a default `Extras` for task cells pushed into a multilevel task
+    /// queue. It generates a random task id and does not specify the fixed
+    /// level.
+    pub fn multilevel_default() -> Extras {
+        Self::new_multilevel(thread_rng().next_u64(), None)
```

Comment:
Why not use UUID or a default atomic counter?

Reply:
Task ID can be passed from the client so an atomic counter seems weird in semantics.

I can change to use UUID. I just thought 64 bits is enough. According to birthday attack, if we store 1e7 random u64 ids, the possibility of id collision is only 2.7e-6.